server/src/lsp-server.ts

```diff
@@ -698,8 +698,17 @@ export class LspServer {
 
         // organize import is provided by tsserver for any line, so we only get it if explicitly requested
         if (params.context.only && params.context.only.includes(CodeActionKind.SourceOrganizeImports)) {
+            // see this issue for more context about how this argument is used
+            // https://github.com/microsoft/TypeScript/issues/43051
+            const skipDestructiveCodeActions = params.context.diagnostics.some(
+                // assume no severity is an error
+                d => (d.severity ?? 0) <= 2
+            );
             actions.push(...provideOrganizeImports(
-                await this.getOrganizeImports({ scope: { type: 'file', args } })
+                await this.getOrganizeImports({
+                    scope: { type: 'file', args },
+                    skipDestructiveCodeActions
+                })
```

Comment:
Do we need to start version-checking here or older versions will accept this argument with no complains?